src/main/java/org/motechproject/prevac/helper/DtoLookupHelper.java

```diff
@@ -107,30 +108,41 @@ public static GridSettings changeLookupForVisitReschedule(GridSettings settings)
             fieldsMap.put(Visit.VISIT_TYPE_PROPERTY_NAME, AVAILABLE_VISIT_TYPES_FOR_RESCHEDULE_SCREEN);
         } else {
             fieldsMap = getFields(settings.getFields());
-            switch (settings.getLookup()) {
-                case "Find By Visit Type":
-                    String type = (String) fieldsMap.get(Visit.VISIT_TYPE_PROPERTY_NAME);
-                    if (StringUtils.isBlank(type) || !AVAILABLE_VISIT_TYPES_FOR_RESCHEDULE_SCREEN.contains(VisitType.valueOf(type))) {
-                        fieldsMap.put(Visit.VISIT_TYPE_PROPERTY_NAME, null);
-                    }
-                    settings.setLookup(settings.getLookup() + " And Planned Date");
-                    break;
-                default:
-                    fieldsMap.put(Visit.VISIT_TYPE_PROPERTY_NAME, AVAILABLE_VISIT_TYPES_FOR_RESCHEDULE_SCREEN);
-                    settings.setLookup(settings.getLookup() + " And Visit Type Set And Planned Date");
-                    break;
-            }
         }
 
         Map<String, String> rangeMap = getDateRangeFromFilter(settings);
-
-        if (rangeMap != null && (StringUtils.isNotBlank(rangeMap.get("min")) || StringUtils.isNotBlank(rangeMap.get("max")))) {
-            settings.setLookup(settings.getLookup() + " Range");
-            fieldsMap.put(Visit.VISIT_PLANNED_DATE_PROPERTY_NAME, rangeMap);
-        } else {
-            fieldsMap.put(Visit.VISIT_PLANNED_DATE_PROPERTY_NAME, null);
+        String lookup = settings.getLookup();
+
+        switch (lookup) {
+            case "Find By Visit Type And Planned Date Range":
+                break;
+            case "Find By Visit Type Set And Planned Date":
+                if (rangeMap != null && (StringUtils.isNotBlank(rangeMap.get("min")) || StringUtils.isNotBlank(rangeMap.get("max")))) {
+                    settings.setLookup(lookup + " Range");
+                    fieldsMap.put(Visit.VISIT_PLANNED_DATE_PROPERTY_NAME, rangeMap);
+                }
+                break;
+            case "Find By Visit Type And Actual Date Range":
+                if (rangeMap != null && (StringUtils.isNotBlank(rangeMap.get("min")) || StringUtils.isNotBlank(rangeMap.get("max")))) {
+                    settings.setLookup(lookup + " And Planned Date Range");
+                    fieldsMap.put(Visit.VISIT_PLANNED_DATE_PROPERTY_NAME, rangeMap);
+                }
+                break;
+            case "Find By Participant Id":
+            case "Find By Participant Name":
+            case "Find By Clinic Location":
+            case "Find By Visit Actual Date Range":
+                if (rangeMap != null && (StringUtils.isNotBlank(rangeMap.get("min")) || StringUtils.isNotBlank(rangeMap.get("max")))) {
```

Comment:
I think you can move it to default, add the logic that is already there in the "else" and add case for "Find By Visit Planned Date Range" instead